streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java

```diff
@@ -212,7 +212,7 @@ private ByteBuffer encodeVersionThreeFourAndFive(final int usedVersion) {
         final ByteBuffer buf = ByteBuffer.allocate(getVersionThreeFourAndFiveByteLength(endPointBytes));
 
         buf.putInt(usedVersion); // used version
-        buf.putInt(LATEST_SUPPORTED_VERSION); // supported version
+        buf.putInt(latestSupportedVersion); // supported version
```

Comment:
This is actually not necessary for `SubscriptionInfo` at this point in time, but may prevent a similar bug in the future.

Reply:
Sounds good. Hopefully we do not need to decode and encode a subscription info object within the same instance :P

Reply:
HahaI guess you never know